testmodule/providers/jsonrpc/testJsonRpcSubscriptions.nim

```diff
@@ -174,3 +178,16 @@ suite "HTTP polling subscriptions - filter not found":
     await subscriptions.unsubscribe(id)
 
     check not subscriptions.subscriptionMapping.hasKey id
+
+  test "polling continues with new filter after temporary error":
+    let filter = EventFilter(address: Address.example, topics: @[array[32, byte].example])
+    let emptyHandler = proc(log: Log) = discard
+
+    let id = await subscriptions.subscribeLogs(filter, emptyHandler)
+
+    await stopServer()
```

Comment:
I tried this as well and the errors I received were not the same errors that were being swallowed in the testnet. I was only able to emulate the exact error by disconnecting the client without stopping it (by connecting it to a closed port).

Reply:
I just tried out connecting to a closed port, like you did [here](https://github.com/codex-storage/nim-ethers/commit/39d03f642cf6fa6e47f21c6aa3d70a728af68823#diff-10d724a619389f6f12a5d2743337651e9bd50084f7b9e7c084d4472f76fba6a1R141), but it gave me the exact same JsonRpcError "Failed to send POST Request with JSON-RPC: Could not connect to remote host" :shrug:

Reply:
My apologies, I remember now why this didn't work for me. I was testing that the subscription would survive across temporary disconnections, and because restarting the mock server required an instantiation of `MockRpcHttpServer.new`, the subscriptions were reset. Thus it was not sufficient for my liking, and why I resorted to connecting to an invalid port. This allowed me to keep the existing `MockRpcHttpServer` instance while restarting the failed polling loop on the client side.